lib/perl/Genome/InstrumentData/AlignmentResult/Command/PicardRnaSeqMetrics.pm

```diff
@@ -78,8 +79,37 @@ sub chart_output_filename {
     return Genome::Model::Build::RnaSeq->default_chart_output_filename;
 }
 
+sub required_files_from_annotation_build {
+    (qw/ rRNA_MT_file annotation_file /);
+}
+
+sub does_annotation_build_have_required_files {
+    my ($self, $annotation_build, $reference_build) = Params::Validate::validate_pos(
+        @_, {isa => __PACKAGE__}, {type => OBJECT}, {type => OBJECT},
+    );
+
+    my @missing_files;
+    for my $file_method ( required_files_from_annotation_build() ) {
+        my $file = $annotation_build->$file_method('gtf', $reference_build->id ,0);
+        if ( not -s $file ) {
+            $file = $annotation_build->$file_method('gtf', undef, 0);
+            if ( not -s $file ) {
+                push @missing_files, $file_method;
+            }
+        }
+    }
+
+    if ( @missing_files ) {
+        die $self->error_message("Cannot proceed! Missing required files from annotation build: @missing_files",);
```

Comment:
Sure. Would be nice if we all agreed on this sort of behavior. I get the 'does' seem like a check, and would not be so harsh as to die if the check failed.